cpu/atmega_common/periph/uart.c

```diff
@@ -74,94 +78,193 @@ static mega_uart_t *dev[] = {
 };
 
 /**
- * @brief   Allocate memory to store the callback functions.
+ * @brief   Allocate memory to store the settings.
  */
 static uart_isr_ctx_t isr_ctx[UART_NUMOF];
+static uint16_t uart_brr[UART_NUMOF];
 
-static void _update_brr(uart_t uart, uint16_t brr, bool double_speed)
+static inline uint8_t _is_asleep(uart_t uart)
 {
-    dev[uart]->BRR = brr;
-    if (double_speed) {
-        dev[uart]->CSRA |= (1 << U2X0);
+#if defined(CPU_ATMEGA328P)
+    (void)uart;
+    return ((PRR >> PRUSART0) & 0x1);
+#else
+
+    if (uart == 0) {
+        return ((PRR0 >> PRUSART0) & 0x1);
+    }
+
+#if defined(CPU_ATMEGA1284P)
```

Comment:
by moving this `#if` inside `else if (uart == 1) {` two lines can be saved and readability is improved a bit

Reply:
Done.